app/[locale]/@modals/(.)search/page.tsx

```diff
@@ -1,10 +1,14 @@
 import Search from '~/app/search/search';
 import { Dialog } from '~/components/dialog';
 
-export default function Page() {
+export default function Page({
+  searchParams,
+}: {
+  searchParams: Record<string, string | undefined>;
```

Comment:
Hmm. Unsure where this type came from, I think it can just be simple:
```suggestion
  searchParams: { q: string };
```